File: Practice/src/coding_basic_training/CreateArray6.js
function solution(arr) {
    var answer = [];
    for (let i of arr) {
        if (answer.length == 0 || answer[answer.length - 1] != i) {
            answer.push(i);
        } else {
            answer.pop();
            // answer.splice(-1);
        }
    }
    return answer.length > 0 ? answer : [-1];
}

File: Practice/src/coding_basic_training/CreateArray6.java
package coding_basic_training;

import java.util.Stack;

public class CreateArray6 {
//	배열 만들기6
	public int[] solution(int[] arr) {
		Stack<Integer> s = new Stack<Integer>();
		for (int i = 0; i < arr.length; i++) {
			if (s.size() == 0 || s.peek() != arr[i]) {
				s.add(arr[i]);
			} else if (s.peek() == arr[i]) {
				s.pop();
			}
		}
		return s.size() > 0 ? s.stream().mapToInt(i -> i).toArray() : new int[] { -1 };
	}
}